packages/rocketchat-channel-settings/client/views/channelSettings.coffee

```diff
@@ -106,4 +98,13 @@ Template.channelSettings.onCreated ->
 								return toastr.error TAPi18n.__(err.reason, err.details.roomType)
 							return toastr.error TAPi18n.__(err.reason)
 						toastr.success TAPi18n.__ 'Room_type_changed_successfully'
+			when 'archivationState'
+				if @$('input[name=archivationState]:checked').val() is 'true'
+					Meteor.call 'archiveRoom', @data?.rid, (err, results) ->
```

Comment:
@graywolf336 What he means is that before calling `archiveRoom` or `unarchiveRoom` there should be a check whether it is even necessary to archive the room or if it is archived already. (i.e. If the user edits the settings of an archived channel, leaves the radio button on archived and clicks on save.)
@rodrigok I'll make sure to add that in.